Required for correct drag/drop handling on Linux. This patch enables events to
correctly pass from the Scintilla GTK widget up to the Mozilla widgets.

Index: widget/src/gtk2/nsWindow.cpp
===================================================================
RCS file: /cvsroot/mozilla/widget/src/gtk2/nsWindow.cpp,v
retrieving revision 1.99.2.1
diff -d -u -r1.99.2.1 nsWindow.cpp
--- widget/src/gtk2/nsWindow.cpp	10 Jun 2004 15:02:36 -0000	1.99.2.1
+++ widget/src/gtk2/nsWindow.cpp	13 Sep 2004 21:13:30 -0000
@@ -2937,7 +2937,7 @@
 
 /* static */
 nsWindow *
-get_window_for_gdk_window(GdkWindow *window)
+_get_window_for_gdk_window(GdkWindow *window)
 {
     gpointer user_data;
     user_data = g_object_get_data(G_OBJECT(window), "nsWindow");
@@ -2948,6 +2948,18 @@
     return NS_STATIC_CAST(nsWindow *, user_data);
 }
 
+nsWindow *
+get_window_for_gdk_window(GdkWindow *gdk_window)
+{
+    nsWindow *window = NULL;
+    while (gdk_window) {
+        window = _get_window_for_gdk_window(gdk_window);
+        if (window) break;
+        gdk_window = gdk_window_get_parent(gdk_window);
+    }
+    return window;
+}
+
 /* static */
 nsWindow *
 get_owning_window_for_gdk_window(GdkWindow *window)
